tensorflow/python/platform/googletest.py

```diff
@@ -51,43 +50,9 @@
 # pylint: disable=invalid-name
 # pylint: disable=undefined-variable
 def g_main(argv):
-  """Delegate to absltest.main after redefining testLoader."""
-  if 'TEST_SHARD_STATUS_FILE' in os.environ:
-    try:
-      f = None
-      try:
-        f = open(os.environ['TEST_SHARD_STATUS_FILE'], 'w')
-        f.write('')
-      except IOError:
-        sys.stderr.write('Error opening TEST_SHARD_STATUS_FILE (%s). Exiting.'
-                         % os.environ['TEST_SHARD_STATUS_FILE'])
-        sys.exit(1)
-    finally:
-      if f is not None: f.close()
-
-  if ('TEST_TOTAL_SHARDS' not in os.environ or
-      'TEST_SHARD_INDEX' not in os.environ):
-    return absltest_main(argv=argv)
-
-  total_shards = int(os.environ['TEST_TOTAL_SHARDS'])
-  shard_index = int(os.environ['TEST_SHARD_INDEX'])
-  base_loader = TestLoader()
-
-  delegate_get_names = base_loader.getTestCaseNames
-  bucket_iterator = itertools.cycle(range(total_shards))
-
-  def getShardedTestCaseNames(testCaseClass):
-    filtered_names = []
-    for testcase in sorted(delegate_get_names(testCaseClass)):
-      bucket = next(bucket_iterator)
-      if bucket == shard_index:
-        filtered_names.append(testcase)
-    return filtered_names
-
-  # Override getTestCaseNames
-  base_loader.getTestCaseNames = getShardedTestCaseNames
-
-  absltest_main(argv=argv, testLoader=base_loader)
+  """Delegate to absltest.main."""
+
+  absltest_main(argv=argv)
```

Comment:
The function `g_main` can now be removed completely.

Reply:
Good catch, thanks!

@markdryan I hate to ask but can you recreate this PR with the g_main removal and the import removal as your own commits? I'm not entirely sure what went wrong in our infrastructure but I know this file in particular is a weird one as its implemented as a patch over an internal source file. (If this still fails to go through I can make the change on the inside, but I want you to get author credit if we can!)

Reply:
@rrevan Will do.  Should be done within the next hour or so.

Reply:
@rrevan so this took longer than expected.  The reason is that I'm unable to get things to work without the g_main function.  Removing it and passing absltest_main to app.run does not work as the absltest_main function is called with the wrong arguments if you don't first pass through the g_main function.  With the wrong arguments, it doesn't run any tests, which is even worse than what we currently have.  So I'm just going to push the two patches that I am able to get to work in a separate pull request.  At least this will fix sharding.  Perhaps someone with better knowledge of python than I can eliminate g_main in the future.  Of course, if you have some idea of how to eliminate g_main, let me know and I can update the new PR.